appinventor/components/src/com/google/appinventor/components/runtime/BaseAiComponent.java

```diff
@@ -0,0 +1,297 @@
+import java.util.Map;
+import java.util.zip.ZipEntry;
+import java.util.HashMap;
+import org.json.JSONArray;
+import org.json.JSONException;
+
+
+@SimpleObject
+public abstract class BaseAiComponent extends AndroidNonvisibleComponent {
+
+    protected static final String LOG_TAG = BaseAiComponent.class.getSimpleName();
+    private static final String MODEL_PATH_SUFFIX = ".mdl";
+    
+    private static String TRANSFER_MODEL_PREFIX = null;
+    private static String PERSONAL_MODEL_PREFIX = null;
+
+    private static final int ERROR_WEBVIEWER_REQUIRED = -7;
+    private static final int ERROR_CLASSIFICATION_FAILED = -2;
+    private static final int ERROR_INVALID_MODEL_FILE = -8;
+    private static final int ERROR_MODEL_REQUIRED = -9;
+    private static final String ERROR_WEBVIEWER_NOT_SET =
+    "You must specify a WebViewer using the WebViewer designer property before you can call %1s";
+
+    private String modelPath = null;
+    private String assetPath = null;
+    private String jsInterface = null;
+    protected WebView webview = null;
+    protected List<String> labels = Collections.emptyList();
+    
+
+    protected BaseAiComponent(Form form) {
+        super(form);
+        requestHardwareAcceleration(form);
+        WebView.setWebContentsDebuggingEnabled(true);
+    }
+    
+   
+    @SuppressLint("SetJavaScriptEnabled")
+    private void configureWebView(WebView webview) {
+      this.webview = webview;
+      webview.getSettings().setJavaScriptEnabled(true);
+      webview.getSettings().setMediaPlaybackRequiresUserGesture(false);
+      // adds a way to send strings to the javascript
+      webview.addJavascriptInterface(new JsObject(), jsInterface);
+      if(jsInterface.equals("PersonalImageClassifier")){
+          TRANSFER_MODEL_PREFIX="https://appinventor.mit.edu/personal-image-classifier/transfer/";
+          PERSONAL_MODEL_PREFIX="https://appinventor.mit.edu/personal-image-classifier/personal/";
+      }else if(jsInterface.equals("PersonalAudioClassifier")){
+          TRANSFER_MODEL_PREFIX="https://appinventor.mit.edu/personal-audio-classifier/transfer/";
+          PERSONAL_MODEL_PREFIX="https://appinventor.mit.edu/personal-audio-classifier/personal/";
+      }
+      webview.setWebViewClient(new WebViewClient() {
+        @Override
+        public WebResourceResponse shouldInterceptRequest(WebView view, String url) {
+          Log.d(LOG_TAG, "shouldInterceptRequest called");
+          InputStream file = null;
+          String charSet;
+          String contentType;
+
+          if (url.endsWith(".json")) {
+            contentType = "application/json";
+            charSet = "UTF-8";
+          } else {
+            contentType = "application/octet-stream";
+            charSet = "binary";
+          }
+
+          try {
+            if (url.contains(TRANSFER_MODEL_PREFIX)) {
+              Log.d(LOG_TAG, "overriding " + url);
+
+              file = form.openAsset(url.substring(TRANSFER_MODEL_PREFIX.length()));
```

Comment:
So this line I think is an issue. You will want to fully qualify the asset using "file://android_asset/" and then the filename. Otherwise, the companion app will try to load it from the file system rather than the packaged assets.